src/lib/dnssd/Advertiser.h

```diff
@@ -116,7 +123,7 @@ class BaseAdvertisingParams
     size_t mMacLength                = 0;
     Optional<ReliableMessageProtocolConfig> mLocalMRPConfig;
     Optional<bool> mTcpSupported;
-    Optional<bool> mICDOperatesAsLIT;
+    ICDModeAdvertise mICDModeAdvertise;
```

Comment:
Shouldn't this default to kNone, and then we don't have to add those explicit sets various places.

Otherwise this is an API change that will silently produce random values for consumers that never set this member.

Reply:
Yes, I can do that